model/histogram/float_histogram.go

```diff
@@ -321,24 +381,45 @@ func (h *FloatHistogram) Add(other *FloatHistogram) *FloatHistogram {
 	h.PositiveSpans, h.PositiveBuckets = addBuckets(h.Schema, h.ZeroThreshold, false, hPositiveSpans, hPositiveBuckets, otherPositiveSpans, otherPositiveBuckets)
 	h.NegativeSpans, h.NegativeBuckets = addBuckets(h.Schema, h.ZeroThreshold, false, hNegativeSpans, hNegativeBuckets, otherNegativeSpans, otherNegativeBuckets)
 
-	return h
+	h.CustomBounds = nil
```

Comment:
I think it's enough to ensure these in constructors/copy constructors. Here it feels a bit too much and also will be hard to ensure we zero out everything, suggest to just ignore those fields.